contracts/test/integration/post-proposal-checks/IntegrationTestRateLimiters.sol

```diff
@@ -0,0 +1,280 @@
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import {VoltV2} from "../../../volt/VoltV2.sol";
+import {CoreV2} from "../../../core/CoreV2.sol";
+import {PCVOracle} from "../../../oracle/PCVOracle.sol";
+import {SystemEntry} from "../../../entry/SystemEntry.sol";
+import {ERC20Allocator} from "../../../pcv/utils/ERC20Allocator.sol";
+import {NonCustodialPSM} from "../../../peg/NonCustodialPSM.sol";
+import {GlobalRateLimitedMinter} from "../../../limiter/GlobalRateLimitedMinter.sol";
+import {GlobalSystemExitRateLimiter} from "../../../limiter/GlobalSystemExitRateLimiter.sol";
+import {PegStabilityModule} from "../../../peg/PegStabilityModule.sol";
+
+contract IntegrationTestRateLimiters is PostProposalCheck {
+    using SafeCast for *;
+
+    address public constant user = 0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7;
+    uint256 public snapshotAfterMints;
+
+    CoreV2 private core;
+    VoltV2 private volt;
+    SystemEntry private systemEntry;
+    ERC20Allocator private allocator;
+    PegStabilityModule private usdcpsm;
+    PegStabilityModule private daipsm;
+    NonCustodialPSM private usdcncpsm;
+    NonCustodialPSM private daincpsm;
+    IERC20 private dai;
+    IERC20 private usdc;
+    GlobalRateLimitedMinter private grlm;
+    GlobalSystemExitRateLimiter private gserl;
+    PCVOracle private pcvOracle;
+    address private morphoUsdcPCVDeposit;
+    address private morphoDaiPCVDeposit;
+
+    function setUp() public override {
+        super.setUp();
+
+        core = CoreV2(addresses.mainnet("CORE"));
+        volt = VoltV2(addresses.mainnet("VOLT"));
+        systemEntry = SystemEntry(addresses.mainnet("SYSTEM_ENTRY"));
+        allocator = ERC20Allocator(addresses.mainnet("PSM_ALLOCATOR"));
+        usdcpsm = PegStabilityModule(addresses.mainnet("PSM_USDC"));
+        daipsm = PegStabilityModule(addresses.mainnet("PSM_DAI"));
+        usdcncpsm = NonCustodialPSM(addresses.mainnet("PSM_NONCUSTODIAL_USDC"));
+        daincpsm = NonCustodialPSM(addresses.mainnet("PSM_NONCUSTODIAL_DAI"));
+        dai = IERC20(addresses.mainnet("DAI"));
+        usdc = IERC20(addresses.mainnet("USDC"));
+        grlm = GlobalRateLimitedMinter(
+            addresses.mainnet("GLOBAL_RATE_LIMITED_MINTER")
+        );
+        gserl = GlobalSystemExitRateLimiter(
+            addresses.mainnet("GLOBAL_SYSTEM_EXIT_RATE_LIMITER")
+        );
+        pcvOracle = PCVOracle(addresses.mainnet("PCV_ORACLE"));
+        morphoUsdcPCVDeposit = addresses.mainnet("PCV_DEPOSIT_MORPHO_USDC");
+        morphoDaiPCVDeposit = addresses.mainnet("PCV_DEPOSIT_MORPHO_DAI");
+    }
+
+    /*
+    Flow of the first user that mints VOLT in the new system.
+    Performs checks on the global rate limits, and accounting
+    in the new system's PCV Oracle.
+    */
+    function testUserPSMMint() public {
+        // read initial buffer left
+        uint256 bufferCap = grlm.bufferCap();
+        uint256 initialBuffer = grlm.buffer();
```

Comment:
we should also add assertions here that the global system exit rate limiter buffer is replenished once skim has been called and funds are moved to the PCV Deposit

Reply:
ok